modules/standard/AutoGpu.chpl

```diff
@@ -28,13 +29,68 @@ module AutoGpu {
   use ChplConfig;
   use Errors;
 
+  /*
+   This attribute can be applied to loops and variable declarations to ensure
+   that they are elgibile for GPU execution. Unlike :annotation:`@assertOnGpu`,
+   this attribute has no runtime effect. It only asserts that the code
+   `could` be executed on the GPU, and not that it `will` be executed.
+
+   When applied to loops, this attribute asserts that the loop itself is
+   GPU-eligible. When applied to variables, this attribute asserts that
+   all promoted expressions and loop expressions (e.g., foreach expressions)
+   are GPU-eligible instead.
```

Comment:
I wonder if you can completely drop variable application of this attribute. That feels more about the parser details rather than what the user cares about. Where, I think this is all about loops from the user's perspective and not really related to variables in any way (it doesn't check if the variable is eligible to be on the GPU, for example).

I know that not all loop expressions are supported, but neither are all variable declarations.

```chpl
@gpu.assertEligible
var A = foo();
```